backend/private-graph/graph/resolver.go

```diff
@@ -298,14 +300,23 @@ func (r *Resolver) isWhitelistedAccount(ctx context.Context) bool {
 	return isAdmin || uid == WhitelistedUID || isDockerDefaultAccount
 }
 
-func (r *Resolver) isDemoProject(project_id int) bool {
-	return project_id == 0
+func (r *Resolver) isDemoProject(ctx context.Context, project_id int) bool {
+	return project_id == r.demoProjectID(ctx)
 }
 
 func (r *Resolver) isDemoWorkspace(workspace_id int) bool {
 	return workspace_id == 0
 }
 
+func (r *Resolver) demoProjectID(ctx context.Context) int {
+	if demoProjectID, err := strconv.Atoi(DemoProjectID); err != nil {
+		log.WithContext(ctx).Error(err, "error converting DemoProjectID to int")
+		return 0
+	} else {
```

Comment:
if `DEMO_PROJECT_ID` is not set, should we `return 0, e.New('')` here and have `isDemoProject` return false? otherwise, this logic implies that if it is not set, we fallback to demo project 0

Reply:
Yeah, I guess I figured `0` would be the same as disabled since there would be no project with that ID. I decided to log rather than erroring - do you think that's appropriate? Happy to change how this works if you think there is a better way to handle!

Reply:
given the comment now, i think this is fine! there isn't a way for a project to ever be 0 so it wouldn't ever match.